refactor(linter): Adopt DefaultRuleConfig in a lot of files

connorshea opened the pull request:
- Update a _lot_ of rules to use the DefaultRuleConfig pattern.- Move code around in a few files to ensure the rule files have similar patterns in terms of order-of-methods in each rule. (e.g. `numeric_separators_style.rs`)
- Fix the docs for `no-extraneous-class` so they have correct casing, previously it was using `serde(renameAll = "camelCase")`, which does not exist.

See also #16285 for a few fixes that were split from this PR.

camc314 commented:
### Merge activity

* **Nov 30, 4:33 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 30, 4:33 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
refactor(linter): Adopt DefaultRuleConfig in a lot of files (#16284)

- Update a _lot_ of rules to use the DefaultRuleConfig pattern.- Move code around in a few files to ensure the rule files have similar patterns in terms of order-of-methods in each rule. (e.g. `numeric_separators_style.rs`)
- Fix the docs for `no-extraneous-class` so they have correct casing, previously it was using `serde(renameAll = "camelCase")`, which does not exist.

See also #16285 for a few fixes that were split from this PR.
